src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/api/DocumentController.kt

```diff
@@ -0,0 +1,127 @@
+package org.vaccineimpact.orderlyweb.controllers.api
+
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.FileSystem
+import org.vaccineimpact.orderlyweb.Files
+import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.db.DocumentRepository
+import org.vaccineimpact.orderlyweb.db.OrderlyDocumentRepository
+import org.vaccineimpact.orderlyweb.models.Document
+import java.io.File
+
+class DocumentController(context: ActionContext,
+                         private val fileSystem: FileSystem,
+                         config: Config,
+                         private val repo: DocumentRepository): Controller(context)
+{
+    constructor(context: ActionContext) :
+            this(context,
+                    Files(),
+                    AppConfig(),
+                    OrderlyDocumentRepository())
+
+    private val topLevelFolder = fileSystem.getAbsolutePath(config["documents.root"])
+
+    fun refreshDocuments(): String {
+        val allDocs = repo.getAllFlat()
+
+        val unrefreshedDocs = allDocs.toMutableList()
+
+        val topLevelDir = unrefreshedDocs.find{it.path == relativePath(topLevelFolder)}
+        if ( topLevelDir == null)
+        {
+            repo.add(relativePath(topLevelFolder), documentName(topLevelFolder), false, null)
```

Comment:
Oh, just have everything directly in the root be parentless?

Reply:
yeah, like the root nodes are the first level items in the dropbox folder, rather than the dropbox folder itself (since that's just there as a container, rather than a part of the deliberately structured file system)